### ISSUE TITLE
How to add/ remove column using button when colspan rowrpan in header? 

### ISSUE THREAD

**Issue by sunhk25** [wenzhixin/bootstrap-table#5656]
**Description**  

Add new column is OK, but it give me some errors when remove the column.

**Example (optional)**  

this is online example
https://live.bootstrap-table.com/code/sunhk25/7013

**Comment by djhvscf** [comment]
FieldIndex is not working properly

**Comment by ko22009** [comment]
@sunhk25, I think It is not good idea destroy and create table anew

**Comment by djhvscf** [comment]
If you use the refreshOption method is not working either. https://live.bootstrap-table.com/code/djhvscf/7030

**Comment by ko22009** [comment]
@djhvscf, I think need fix refreshOptions method better than create extension.

**Comment by sunhk25** [comment]
@ko22009  @djhvscf 
Thank you,  use the refreshOption is smart.
But is still not  working.

**Comment by djhvscf** [comment]
what kind of fix? @ko22009

**Comment by ko22009** [comment]
@djhvscf, need replace algorithm storing of data table, cells and rows generated not correctly with colspan and rowspan.

**Comment by ko22009** [comment]
If you want without problem remove column, you need manually fix columns template.
Example: https://live.bootstrap-table.com/code/ko22009/7044
First: remove element,
Second: add new element.

After that you cannot repeat remove element, bcz now It is not correct html markup.

**Comment by wenzhixin** [comment]
I don't think it should be a bug, at the beginning of the design, we don't think about this case:
```js
<thead>
    <tr>
        <th data-field="id" colspan="1" rowspan="2" data-valign="middle">Item ID</th>
        <th colspan="1">Item Detail</th>
    </tr>
    <tr>
        <th data-field="name" colspan="1">Item Name</th>
    </tr>
</thead>
```

The `colspan` for all three fields is `1`, in this case, there is no way to achieve it.

Working example: https://live.bootstrap-table.com/code/wenzhixin/7048
Workaround example for your case: https://live.bootstrap-table.com/code/wenzhixin/7047

**Comment by ko22009** [comment]
@wenzhixin, we have a bug, if we have data attribute in html code. But if we push to bootstrapTable, the problem disappears:
https://live.bootstrap-table.com/code/ko22009/7049

**Comment by antonioaltamura** [comment]
The ticket https://github.com/wenzhixin/bootstrap-table/issues/5637 has been closed but the problem is still there, is there a workaround in my case?

**Comment by wenzhixin** [comment]
@ko22009 Please help verify that the issue has been fixed in the latest version?
ref #6124 after v1.20.1
